src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBinding.cs

```diff
@@ -77,19 +95,35 @@ public ParameterDescriptor ToParameterDescriptor()
         public Type TriggerValueType => typeof(SignalRTriggerEvent);
 
         // TODO: Use dynamic contract to deal with parameterName
-        public IReadOnlyDictionary<string, Type> BindingDataContract => CreateBindingContract();
+        public IReadOnlyDictionary<string, Type> BindingDataContract { get; }
 
         /// <summary>
         /// Defined what other bindings can use and return value.
         /// </summary>
-        private IReadOnlyDictionary<string, Type> CreateBindingContract()
+        private IReadOnlyDictionary<string, Type> CreateBindingContract(SignalRTriggerAttribute attribute, ParameterInfo parameter)
         {
             var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)
             {
-                //TODO: Add names in parameterNames to contract for binding
                 { ReturnParameterKey, typeof(object).MakeByRefType() },
             };
 
+            // Add names in ParameterNames to binding contract, that user can bind to Functions' parameter directly
+            if (attribute.ParameterNames != null)
```

Comment:
should we check unique? e.g.: `["a", "a"]`

Reply:
I put it in trigger validation https://github.com/Azure/azure-functions-signalrservice-extension/pull/95/files#diff-4af6e37bf8e81919be0c0013885e1bd2R90
It should be ignore case unique